### PR TITLE
Remove `RequiresPreviewFeatures` attribute on `Lock`, add test using the `lock` keyword

### PR DESCRIPTION
**Pull request by kouvel**
- Also removed the `EnablePreviewFeatures` project properties that were added with `Lock` changes
- Added some basic tests with `Lock` using the `lock` keyword

### CODE REVIEW

**Review by mangod9 — APPROVED** [review]

File: src/libraries/System.Runtime/ref/System.Runtime.cs [hunk 1]
```diff
@@ -15158,7 +15158,6 @@ public enum LazyThreadSafetyMode
         PublicationOnly = 1,
         ExecutionAndPublication = 2,
     }
-    [System.Runtime.Versioning.RequiresPreviewFeaturesAttribute]
```
Comment (stephentoub) [inline src/libraries/System.Runtime/ref/System.Runtime.cs:15161]: Are we ready to start using Lock in runtime? Would the general philosophy be to use it anywhere we're currently using a dedicated lock object and it's only being used for enter/exit and not signaling?

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Runtime/ref/System.Runtime.cs (hunk 1 above)
Comment (mangod9) [reply src/libraries/System.Runtime/ref/System.Runtime.cs:15161]: yeah we were planning to initially start using it in Threadpool synchronization logic. But might be good to start using in few other places too, so we can iron out issues (if any) before going broader.

Comment (kouvel) [reply src/libraries/System.Runtime/ref/System.Runtime.cs:15161]: Yea we were planning on using it in a few places relevant to the thread pool to start with. Those locks are used frequently but aren't very contentious. It is already used in NativeAOT, wanted to get some more coverage in CoreCLR as well. It should be ready to be used anywhere Monitor is used currently just as a lock. There are some SOS diagnostics to add, but it should be fine to update existing uses.

Comment (stephentoub) [reply src/libraries/System.Runtime/ref/System.Runtime.cs:15161]: Perf wise, how should I think about it? It'll be as-good-or-better in pretty much any scenario, such that I can use it without doing significant perf validation, or it's more nuanced?

**Pull request merged**